crates/next-core/js/src/entry/config/next.js

```diff
@@ -1,13 +1,16 @@
 import loadConfig from "next/dist/server/config";
 import { PHASE_DEVELOPMENT_SERVER } from "next/dist/shared/lib/constants";
+import loadCustomRoutes from "next/dist/lib/load-custom-routes";
 
 const loadNextConfig = async () => {
   const nextConfig = await loadConfig(PHASE_DEVELOPMENT_SERVER, process.cwd());
 
   nextConfig.generateBuildId = await nextConfig.generateBuildId?.();
-  nextConfig.headers = await nextConfig.headers?.();
-  nextConfig.redirects = await nextConfig.redirects?.();
-  nextConfig.rewrites = await nextConfig.rewrites?.();
+
+  const { headers, redirects, rewrites } = await loadCustomRoutes(nextConfig);
+  nextConfig.headers = headers;
+  nextConfig.redirects = redirects;
+  nextConfig.rewrites = rewrites;
```

Comment:
This should be handled on the Next.js side

Reply:
Removed.